packages/pubsub/src/index.js

```diff
@@ -78,27 +69,139 @@ function PubSub(options) {
     return new PubSub(options);
   }
 
-  this.defaultBaseUrl_ = 'pubsub.googleapis.com';
-  this.determineBaseUrl_(options.apiEndpoint);
-
-  var config = {
-    baseUrl: this.baseUrl_,
-    customEndpoint: this.customEndpoint_,
-    service: 'pubsub',
-    apiVersion: 'v1',
-    scopes: [
-      'https://www.googleapis.com/auth/pubsub',
-      'https://www.googleapis.com/auth/cloud-platform'
-    ],
-    packageJson: require('../package.json')
-  };
-
-  this.options = options;
+  this.options = extend({
+    scopes: v1.ALL_SCOPES
+  }, options);
 
-  commonGrpc.Service.call(this, config, options);
+  this.api = {};
+  this.auth = googleAuth(this.options);
+  this.projectId = this.options.projectId || '{{projectId}}';
 }
 
-util.inherits(PubSub, commonGrpc.Service);
+/**
+ * Create a subscription to a topic.
+ *
+ * All generated subscription names share a common prefix, `autogenerated-`.
+ *
+ * @resource [Subscriptions: create API Documentation]{@link https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/create}
+ *
+ * @throws {Error} If a Topic instance or topic name is not provided.
+ * @throws {Error} If a subName is not provided.
+ *
+ * @param {module:pubsub/topic|string} topic - The Topic to create a
+ *     subscription to.
+ * @param {string=} subName - The name of the subscription. If a name is not
+ *     provided, a random subscription name will be generated and created.
+ * @param {object=} options - See a
+ *     [Subscription resource](https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions)
+ * @param {number} options.ackDeadlineSeconds - The maximum time after receiving
+ *     a message that you must ack a message before it is redelivered.
+ * @param {string} options.encoding - When pulling for messages, this type is
+ *     used when converting a message's data to a string. (default: 'utf-8')
+ * @param {number|date} options.messageRetentionDuration - Set this to override
+ *     the default duration of 7 days. This value is expected in seconds.
+ *     Acceptable values are in the range of 10 minutes and 7 days.
```

Comment:
Maybe @stephenplusplus can chime in here, this is legacy documentation that got shuffled around.